uint8 idx
float32 score
float32 x_3d
float32 y_3d
float32 z_3d
